### PR TITLE
fix(pipelines): Handle collapsed groups better

### LINKED ISSUE #185: Pipeline layout issues when collapsed groups in groups
**Describe the problem**
There are issues with the pipeline topology layout when there are groups inside of groups and the groups are collapsible.

**How do you reproduce the problem?**
Create a complex graph that has groups inside of groups inside of groups that can be collapsed.

**Expected behavior**
The graph lays out the collapsed groups correctly.

**Is this issue blocking you?**
Yes

**Screenshots**
image

**What is your product and what release date are you targeting?**
RHOAI 2.10

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
## What
Closes #185

## Description
Fixes:
  Collapsed pipeline groups position when they are nested two levels deep
  Put spacer nodes created for pipeline group layout into the groups they belong, new helper function `addSpacerNodes`
  Scaled node position and anchors for pipeline layouts
  Export some missing components so they don't need to be specifically imported
  Respect the parameter `labelPosition` when passed to `DefaultTaskGroup`

## Type of change
- [ ] Feature
- [x] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review
image

### CODE REVIEW

**Review by jenny-s51 — COMMENTED** [review]
These changes are looking fantastic @jeff-phillips-18 ! 

Wondering if we would want to pass in arrow indicators on the edges to show the runAfter tasks and better match RHOAI... WDYT?
image

Upper nesting levels look awesome, though for some reason your changes aren't getting applied to the deeper levels:

Screenshot 2024-05-03 at 3 54 46 PM

**Comment by jeff-phillips-18** [comment]
> Wondering if we would want to pass in arrow indicators on the edges to show the runAfter tasks and better match RHOAI... WDYT?
I can add the arrows though to me the dependencies are inferred.

> Upper nesting levels look awesome, though for some reason your changes aren't getting applied to the deeper levels:
The edges that look "off" are due to dependencies of nodes of one group on nodes of another group. The group layout will not consider these dependencies. I will update the data in the demo app to not have such dependencies.

File: packages/demo-app-ts/src/demos/pipelineGroupsDemo/createDemoPipelineGroupsNodes.ts [hunk 1]
```diff
@@ -329,6 +330,739 @@ export const createExecution3 = (runAfter?: string): [string, PipelineNodeModel[
...
+        'feature-attribution',
+        'finalize-eval-quantile-parameters',
+        'get-predictions-column',
+        'batchpredictionjob',
+        'bigquery_output_table',
+        'gcs_output_directory',
+        'model-batch-explanation',
+        'batchpredictionjob',
+        'bigquery_output_table',
+        'gcs_output_directory',
```
Comment (jeff-phillips-18) [inline packages/demo-app-ts/src/demos/pipelineGroupsDemo/createDemoPipelineGroupsNodes.ts:519]: Duplicate of children above (line 512-514)

File: packages/demo-app-ts/src/demos/pipelineGroupsDemo/createDemoPipelineGroupsNodes.ts [hunk 2]
```diff
@@ -329,6 +330,739 @@ export const createExecution3 = (runAfter?: string): [string, PipelineNodeModel[
...
+      id: 'condition-5',
+      label: 'should_run_model_evaluation',
+      type: 'EXECUTION_TASK_NODE',
+      group: true,
+      collapsed: true,
+      width: GROUP_TASK_WIDTH,
+      height: DEFAULT_TASK_HEIGHT,
+      runAfterTasks: ['automl-forecasting-ensemble-2', 'model-upload-2'],
+      children: [
+        'feature_attributions',
```
Comment (jeff-phillips-18) [inline packages/demo-app-ts/src/demos/pipelineGroupsDemo/createDemoPipelineGroupsNodes.ts:795]: This is also a child of `condition-3`

File: packages/demo-app-ts/src/demos/pipelineGroupsDemo/createDemoPipelineGroupsNodes.ts [hunk 3]
```diff
@@ -329,6 +330,739 @@ export const createExecution3 = (runAfter?: string): [string, PipelineNodeModel[
...
+        'feature-attribution-2',
+        'finalize-eval-quantile-parameters-2',
+        'get-predictions-column-2',
+        'batchpredictionjob',
+        'bigquery_output_table',
+        'gcs_output_directory',
```
Comment (jeff-phillips-18) [inline packages/demo-app-ts/src/demos/pipelineGroupsDemo/createDemoPipelineGroupsNodes.ts:801]: These are also children of `condition-3`

File: packages/demo-app-ts/src/demos/pipelineGroupsDemo/createDemoPipelineGroupsNodes.ts [hunk 4]
```diff
@@ -329,6 +330,739 @@ export const createExecution3 = (runAfter?: string): [string, PipelineNodeModel[
...
+        'model-batch-explanation-2',
+        'batchpredictionjob',
+        'bigquery_output_table',
+        'gcs_output_directory',
```
Comment (jeff-phillips-18) [inline packages/demo-app-ts/src/demos/pipelineGroupsDemo/createDemoPipelineGroupsNodes.ts:805]: These are duplicated above and also children of `condition-3`

**Review by jessiehuff — APPROVED** [review]

**Review by jenny-s51 — APPROVED** [review]

**Pull request merged**